From 1d7c2f69f5f4f1faf0870d5b5c99b8d6a508ff5a Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sun, 30 May 2021 14:30:56 +0200
Subject: [PATCH] Replace werkzeug.posixemulation.rename with os.replace

This function has been always internal, and has been removed from
Werkzeug 2.x.

Since 3.3, Python provides os.replace, a cross-platform atomic rename
function.

Ref: https://github.com/pallets/werkzeug/issues/1759
Ref: https://bugs.python.org/issue8828
---
 lektor/builder.py |    3 +--
 lektor/utils.py   |    3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

--- a/lektor/builder.py
+++ b/lektor/builder.py
@@ -12,7 +12,6 @@ from itertools import chain
 from collections import deque, namedtuple
 
 import click
-from werkzeug.posixemulation import rename
 
 from lektor._compat import PY2, iteritems, text_type
 from lektor.context import Context
@@ -824,7 +823,7 @@ class Artifact(object):
                 op(con)
 
             if self._new_artifact_file is not None:
-                rename(self._new_artifact_file, self.dst_filename)
+                os.replace(self._new_artifact_file, self.dst_filename)
                 self._new_artifact_file = None
 
             if con is not None:
--- a/lektor/utils.py
+++ b/lektor/utils.py
@@ -30,7 +30,6 @@ from markupsafe import Markup
 from slugify import slugify as _slugify
 from werkzeug import urls
 from werkzeug.http import http_date
-from werkzeug.posixemulation import rename
 from werkzeug.urls import url_parse
 
 from lektor._compat import (queue, integer_types, iteritems, reraise,
@@ -492,7 +491,7 @@ def atomic_open(filename, mode='r'):
     else:
         f.close()
         if tmp_filename is not None:
-            rename(tmp_filename, filename)
+            os.replace(tmp_filename, filename)
 
 
 def portable_popen(cmd, *args, **kwargs):
